### DISCUSSION TITLE
my Sample Chart

### DISCUSSION THREAD

**Discussion by d1etmar** [shivan/homeassistant-liquidcheck#37] [Dashboard Ideas and Reporting]
His is my chart ...

<img width="469" height="673" alt="grafik" src="https://github.com/user-attachments/assets/2de5cfc9-2e1d-4b94-a733-22f1902ab189" />

```
type: custom:stack-in-card
title: null
mode: vertical
card_mod:
  style: |
    ha-card { border: none; box-shadow: none; }  
cards:
  - type: custom:stack-in-card
    mode: horizontal
    card_mod:
      style: |
        ha-card { border: none; box-shadow: none; }      
    cards:
      - type: entities
        card_mod:
          style: |
            ha-card { border: none; box-shadow: none; }  
        entities:
          - entity: sensor.inhalt_netto
            type: custom:multiple-entity-row
            name: Zisterne
            secondary_info: last-changed
            show_state: false
            tap_action:
              action: call-service
              service: script.start_zisterne_measurement
            entities:
              - entity: sensor.inhalt_netto
                name: Inhalt
                tap_action:
                  action: call-service
                  service: script.start_zisterne_measurement
              - entity: input_number.water_gain
                name: Zugang
              - entity: input_number.water_loss
                name: Abgang
  - type: custom:apexcharts-card
    card_mod:
      style: |
        .apexcharts-marker,
         .apexcharts-tooltip-marker {
            display: none !important;
         }        
        ha-card { border: none; box-shadow: none; }      
        .apexcharts-tooltip {
            align-items: center;
          }
    header:
      show: true
      show_states: true
      colorize_states: true
    update_interval: 30m
    graph_span: 10days
    span:
      end: day
    show:
      loading: true
    yaxis:
      - id: first
        decimals: 0
        min: 0
        max: 6000
      - id: second
        opposite: true
        min: 0
        decimals: 0
        max: 40
    apex_config:
      dataLabels:
        enabled: true
        textAnchor: start
      tooltip:
        shared: true
        intersect: false
        x:
          format: dd-MMM
          show: true
        "y":
          formatter: |
            EVAL: function(val1, { series, seriesIndex, dataPointIndex }) {
              return `${series[0][dataPointIndex].toFixed(2)} L Regen<br>` +
                     `${series[1][dataPointIndex].toFixed(2)} L Zisterne<br>` ;
                     
            }
          title:
            formatter: |
              EVAL: function(seriesName) {
                return ""
              }
      chart:
        height: 200px
      grid:
        show: true
        strokeDashArray: 0
        borderColor: "#444444"
        position: back
        xaxis:
          lines:
            show: true
        yaxis:
          lines:
            show: true
      xaxis:
        tooltip:
          enabled: false
        type: datetime
        axisTicks:
          show: true
        labels:
          format: dd-MMM
        lines:
          show: true
    series:
      - entity: sensor.gw2000a_daily_rain
        color: green
        name: Regen
        type: area
        yaxis_id: second
        opacity: 0.7
        invert: false
        stroke_width: 2
        offset: "-0d"
        extend_to: false
        group_by:
          func: max
          duration: 1day
          fill: last
        show:
          datalabels: true
          in_header: false
          in_legend: true
          legend_value: false
          in_chart: true
          offset_in_name: false
      - entity: sensor.192_168_199_156_inhalt
        transform: return x - 550
        color: blue
        name: Zisterne
        opacity: 0.5
        type: column
        invert: false
        stroke_width: 1
        yaxis_id: first
        offset: "-0d"
        extend_to: now
        group_by:
          func: last
          duration: 1day
          fill: last
        show:
          datalabels: false
          in_header: false
          in_legend: true
          legend_value: false
          in_chart: true
          offset_in_name: false
      - entity: sensor.192_168_199_156_inhalt
        transform: return x - 550
        color: orange
        name: Max
        type: line
        invert: false
        stroke_width: 2
        yaxis_id: first
        offset: "-0d"
        extend_to: now
        group_by:
          func: max
          duration: 24hours
          fill: last
        show:
          datalabels: false
          in_header: false
          in_legend: true
          legend_value: false
          in_chart: true
          offset_in_name: false
      - entity: sensor.192_168_199_156_inhalt
        transform: return x - 550
        color: red
        name: Min
        type: line
        invert: false
        stroke_width: 1
        yaxis_id: first
        offset: "-0d"
        extend_to: now
        group_by:
          func: min
          duration: 1day
          fill: last
        show:
          datalabels: false
          in_header: false
          in_legend: true
          legend_value: false
          in_chart: true
          offset_in_name: false
    all_series_config:
      stroke_width: 2
      type: line
  - type: custom:plotly-graph
    card_mod:
      style: |
        ha-card {
          background: var(--ha-card-background, var(--card-background-color, #1c1c1c));
          box-shadow: none;
          border: none;
        }
    fn1: |
      $ex {
        vars.max_liter = 5234;
        vars.current_liter = parseFloat(hass.states['sensor.inhalt_netto']?.state) || 0;
        vars.gain = parseFloat(hass.states['input_number.water_gain']?.state) || 0;
        vars.loss = parseFloat(hass.states['input_number.water_loss']?.state) || 0;
        vars.diff = 5234 - vars.current_liter
        // Füllhöhe berechnen (0 bis 170 für den eckigen Kasten)
        vars.fill_percent = Math.min(Math.max(vars.current_liter / vars.max_liter, 0), 1);
        vars.h = vars.fill_percent * 170;
      }
    entities:
      - entity: sensor.inhalt_netto
        show_value: false
    raw_plotly_config: true
    config:
      staticPlot: true
      displayModeBar: false
    layout:
      height: 280
      autosize: true
      paper_bgcolor: rgba(0,0,0,0)
      plot_bgcolor: rgba(0,0,0,0)
      margin:
        t: 10
        b: 10
        l: 0
        r: 0
      xaxis:
        visible: false
        range:
          - -150
          - 150
        fixedrange: true
      yaxis:
        visible: false
        range:
          - -5
          - 185
        fixedrange: true
      annotations:
        - x: 0
          "y": 182
          text: $ex Math.round(vars.diff) + " l"
          showarrow: false
          font:
            size: 26
            color: white
            weight: 900
        - x: 0
          "y": 105
          text: $ex Math.round(vars.current_liter) + " l"
          showarrow: false
          font:
            size: 26
            color: white
            weight: 900
        - x: 0
          "y": 78
          text: $ex "+ " + vars.gain + " l"
          showarrow: false
          font:
            size: 14
            color: "#4caf50"
            weight: 700
        - x: 0
          "y": 58
          text: $ex "- " + vars.loss + " l"
          showarrow: false
          font:
            size: 14
            color: "#f44336"
            weight: 700
      shapes:
        - type: rect
          x0: -75
          x1: 75
          y0: 0
          y1: $ex vars.h
          fillcolor: "#2a5a8a"
          line:
            width: 0
        - type: rect
          x0: -75
          x1: 75
          y0: 0
          y1: 170
          line:
            color: "#3d3d3d"
            width: 3
```

und hier die configurationen dau :

```
rest_command:
  messung_zisterne:
    url: "http://192.168.199.156/command"
    method: POST
    payload: >
      {
        "header": {
          "namespace": "Device.Control",
          "name": "StartMeasure",
          "messageId": "1",
          "payloadVersion": "1"
        },
        "payload": null
      }
    content_type: "application/json; charset=utf-8"

script:
  start_zisterne_measurement:
    sequence:
      - service: rest_command.messung_zisterne
      - delay: "00:00:10"


# water_tracker.yaml

input_number:
  water_last_value:
    name: Water Last Value
    min: 0
    max: 10000
    step: 1
    mode: box

  water_gain:
    name: Water Gain Today
    min: 0
    max: 10000
    step: 1
    mode: box

  water_loss:
    name: Water Loss Today
    min: 0
    max: 10000
    step: 1
    mode: box

automation:
  - alias: Track Water Reservoir Changes
    description: Track water gain/loss based on reservoir level changes
    trigger:
      - platform: state
        entity_id: sensor.192_168_199_156_inhalt
    condition:
      - condition: template
        value_template: >
            {% set sensor_value = states('sensor.192_168_199_156_inhalt') | float(0) %}
            {% set last_value = states('input_number.water_last_value') | float(0) %}
            {{ sensor_value > 0 and sensor_value != last_value }}
    action:
      - variables:
          current: "{{ states('sensor.192_168_199_156_inhalt') | float(0) }}"
          last: "{{ states('input_number.water_last_value') | float(0) }}"
          delta: "{{ current - last }}"
      - choose:
          - conditions: "{{ delta > 0 }}"
            sequence:
              - service: input_number.set_value
                data:
                  entity_id: input_number.water_gain
                  value: >
                    {{ states('input_number.water_gain') | float + delta }}
          - conditions: "{{ delta < 0 }}"
            sequence:
              - service: input_number.set_value
                data:
                  entity_id: input_number.water_loss
                  value: >
                    {{ states('input_number.water_loss') | float + (-delta) }}
      - service: input_number.set_value
        data:
          entity_id: input_number.water_last_value
          value: "{{ current }}"
    mode: queued

  - alias: Reset Daily Water Gain and Loss
    description: Resets daily water gain and loss values at midnight
    trigger:
      - platform: time
        at: '00:00:00'
    action:
      - service: input_number.set_value
        data:
          entity_id: input_number.water_gain
          value: 0
      - service: input_number.set_value
        data:
          entity_id: input_number.water_loss
          value: 0

template:
  - sensor:
      - name: "Inhalt Netto"
        unique_id: inhalt_netto
        unit_of_measurement: "L" 
        state_class: measurement
        state: >
          {% set original = states('sensor.192_168_199_156_inhalt') | float(0) %}
          {{ original - 550 }}
```

**Comment by d1etmar** [comment]
this does some more things .
* If you klick on the first dashboard card it will start measure .
* I cannot pull about 550 L out of the Zisterne , therefore i made a "netto" sensor .
* Daily gain and loss 
* a lot of differenct hacs cards like apexchart,plotly and so on ... 

All needed to optimize the Water usage . In my case if the Rain sensor say about 10L i gain about 2000L . ( 240qm roof ) .
Therefore i mixxed the daily rain with the fillig of the zisterne ( with min and max per day )

Great to know how much water was needed to sprinke the Garden, and appr how long it will last continue so ....

here more expanded and more days to see it a bit better ...

<img width="1859" height="302" alt="grafik" src="https://github.com/user-attachments/assets/0c3deed4-dbc3-44b2-a501-e1df82119bcd" />


hf

**Comment by d1etmar** [comment]
lol, the apex runs with short term history only , so if you want more than that here is the apexchart with "statistics" to work ...

state of measurement und so :) ...

more history , but also about 1 hour late on updates .

```
type: custom:stack-in-card
title: null
mode: vertical
card_mod:
  style: |
    ha-card { border: none; box-shadow: none; }  
cards:
  - type: custom:stack-in-card
    mode: horizontal
    card_mod:
      style: |
        ha-card { border: none; box-shadow: none; }      
    cards:
      - type: entities
        card_mod:
          style: |
            ha-card { border: none; box-shadow: none; }  
        entities:
          - entity: sensor.inhalt_netto
            type: custom:multiple-entity-row
            name: Zisterne
            secondary_info: last-changed
            show_state: false
            tap_action:
              action: call-service
              service: script.start_zisterne_measurement
            entities:
              - entity: sensor.inhalt_netto
                name: Inhalt
                tap_action:
                  action: call-service
                  service: script.start_zisterne_measurement
              - entity: input_number.water_gain
                name: Zugang
              - entity: input_number.water_loss
                name: Abgang
  - type: custom:apexcharts-card
    card_mod:
      style: |
        .apexcharts-marker,
         .apexcharts-tooltip-marker {
            display: none !important;
         }        
        ha-card { border: none; box-shadow: none; }      
        .apexcharts-tooltip {
            align-items: center;
          }
    header:
      show: true
      show_states: true
      colorize_states: true
    update_interval: 30m
    graph_span: 31days
    show:
      loading: true
    yaxis:
      - id: first
        decimals: 0
        min: 0
        max: 6000
      - id: second
        opposite: true
        min: 0
        decimals: 0
        max: 40
    apex_config:
      dataLabels:
        enabled: true
        textAnchor: start
      tooltip:
        shared: true
        intersect: false
        x:
          format: dd-MMM
          show: true
        "y":
          formatter: |
            EVAL: function(val1, { series, seriesIndex, dataPointIndex }) {
              return `${series[0][dataPointIndex].toFixed(2)} L Regen<br>` +
                     `${series[1][dataPointIndex].toFixed(2)} L Zisterne<br>` ;
                     
            }
          title:
            formatter: |
              EVAL: function(seriesName) {
                return ""
              }
      chart:
        height: 200px
      grid:
        show: true
        strokeDashArray: 0
        borderColor: "#444444"
        position: back
        xaxis:
          lines:
            show: true
        yaxis:
          lines:
            show: true
      xaxis:
        tooltip:
          enabled: false
        type: datetime
        axisTicks:
          show: true
        labels:
          format: dd-MMM
        lines:
          show: true
    series:
      - entity: sensor.gw2000a_daily_rain
        color: green
        name: Regen
        type: area
        yaxis_id: second
        opacity: 0.7
        invert: false
        stroke_width: 2
        offset: "-0d"
        extend_to: false
        statistics:
          type: max
          period: hour
          align: middle
        group_by:
          func: max
          duration: 1day
          fill: last
        show:
          datalabels: true
          in_header: false
          in_legend: true
          legend_value: false
          in_chart: true
          offset_in_name: false
      - entity: sensor.192_168_199_156_inhalt
        transform: return x - 550
        color: blue
        name: Zisterne
        opacity: 0.5
        type: column
        invert: false
        stroke_width: 1
        yaxis_id: first
        offset: "-0d"
        extend_to: false
        statistics:
          type: mean
          period: hour
          align: middle
        group_by:
          func: last
          duration: 1day
          fill: last
        show:
          datalabels: false
          in_header: false
          in_legend: true
          legend_value: false
          in_chart: true
          offset_in_name: false
      - entity: sensor.192_168_199_156_inhalt
        transform: return x - 550
        color: orange
        name: Max
        type: line
        invert: false
        stroke_width: 2
        yaxis_id: first
        offset: "-0d"
        extend_to: false
        statistics:
          type: max
          period: hour
          align: middle
        group_by:
          func: max
          duration: 24hours
          fill: last
        show:
          datalabels: false
          in_header: false
          in_legend: true
          legend_value: false
          in_chart: true
          offset_in_name: false
      - entity: sensor.192_168_199_156_inhalt
        transform: return x - 550
        color: red
        name: Min
        type: line
        invert: false
        stroke_width: 1
        yaxis_id: first
        offset: "-0d"
        extend_to: false
        statistics:
          type: min
          period: hour
          align: middle
        group_by:
          func: min
          duration: 1day
          fill: last
        show:
          datalabels: false
          in_header: false
          in_legend: true
          legend_value: false
          in_chart: true
          offset_in_name: false
    all_series_config:
      stroke_width: 2
      type: line
  - type: custom:plotly-graph
    card_mod:
      style: |
        ha-card {
          background: var(--ha-card-background, var(--card-background-color, #1c1c1c));
          box-shadow: none;
          border: none;
        }
    fn1: |
      $ex {
        vars.max_liter = 5234;
        vars.current_liter = parseFloat(hass.states['sensor.inhalt_netto']?.state) || 0;
        vars.gain = parseFloat(hass.states['input_number.water_gain']?.state) || 0;
        vars.loss = parseFloat(hass.states['input_number.water_loss']?.state) || 0;
        vars.diff = 5234 - vars.current_liter
        // Füllhöhe berechnen (0 bis 170 für den eckigen Kasten)
        vars.fill_percent = Math.min(Math.max(vars.current_liter / vars.max_liter, 0), 1);
        vars.h = vars.fill_percent * 170;
      }
    entities:
      - entity: sensor.inhalt_netto
        show_value: false
    raw_plotly_config: true
    config:
      staticPlot: true
      displayModeBar: false
    layout:
      height: 280
      autosize: true
      paper_bgcolor: rgba(0,0,0,0)
      plot_bgcolor: rgba(0,0,0,0)
      margin:
        t: 10
        b: 10
        l: 0
        r: 0
      xaxis:
        visible: false
        range:
          - -150
          - 150
        fixedrange: true
      yaxis:
        visible: false
        range:
          - -5
          - 185
        fixedrange: true
      annotations:
        - x: 0
          "y": 182
          text: $ex Math.round(vars.diff) + " l"
          showarrow: false
          font:
            size: 26
            color: white
            weight: 900
        - x: 0
          "y": 105
          text: $ex Math.round(vars.current_liter) + " l"
          showarrow: false
          font:
            size: 26
            color: white
            weight: 900
        - x: 0
          "y": 78
          text: $ex "+ " + vars.gain + " l"
          showarrow: false
          font:
            size: 14
            color: "#4caf50"
            weight: 700
        - x: 0
          "y": 58
          text: $ex "- " + vars.loss + " l"
          showarrow: false
          font:
            size: 14
            color: "#f44336"
            weight: 700
      shapes:
        - type: rect
          x0: -75
          x1: 75
          y0: 0
          y1: $ex vars.h
          fillcolor: "#2a5a8a"
          line:
            width: 0
        - type: rect
          x0: -75
          x1: 75
          y0: 0
          y1: 170
          line:
            color: "#3d3d3d"
            width: 3

```
